### ISSUE TITLE
how to generate KeyedJaggedTensor on GPU

### ISSUE THREAD

**Issue by xiexbing** [meta-pytorch/torchrec#1715]
Hello, 

I compressed the values, and lengths to reduce the PCIe cost. then I want to move the keys, compressed values and lengths to GPU, and then decompress and generate KJT on GPUs.

My question is: any examples on how to do this?

thanks.

**Comment by henrylhtsang** [comment]
what do you mean by "compress"? They should be torch.tensors right?

**Comment by xiexbing** [comment]
Yes, My question is: can I generate KJT on GPUs directly? The reason I want
to do so is I want to reduce the data size transferred via PCIe, I need to
convert the compressed tensors to the values/lengths tensors as KJT format.

**Comment by xiexbing** [comment]
sorry, I have keys as a string list, values as a list tensor, lengths as a list tensor.

**Comment by henrylhtsang** [comment]
@xiexbing I think you can just first move the tensors to GPU, and then initialize a kjt that way? if I remember correctly, that should work just fine?
